app/src/main/java/com/epfl/drawyourpath/userProfile/cache/UserEntity.kt

```diff
@@ -0,0 +1,157 @@
+package com.epfl.drawyourpath.userProfile.cache
+
+import android.annotation.SuppressLint
+import android.content.res.Resources
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import androidx.core.graphics.drawable.toBitmap
+import androidx.room.ColumnInfo
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+import com.epfl.drawyourpath.R
+import java.io.ByteArrayOutputStream
+import java.time.LocalDate
+import java.time.temporal.ChronoUnit
+
+@Entity(tableName = "User")
+data class UserEntity(
+    /**
+     * the id of the user
+     */
+    @PrimaryKey
+    @ColumnInfo(name = "id")
+    val userId: String,
+
+    /**
+     * the username of the user
+     */
+    val username: String = "",
+
+    /**
+     * the email address of the user
+     */
+    @ColumnInfo(name = "email_address")
+    val emailAddress: String = "",
+
+    /**
+     * the firstname of the user
+     */
+    val firstname: String = "",
+
+    /**
+     * the surname of the user
+     */
+    val surname: String = "",
+
+    /**
+     * the date of birth of the user

```

Comment:
yes, it is normal, it is only useful to specify the name when we want it to be different than the variable name. It is only because it differs in the naming conventions as for example : dateOfBirth is in camelCase as variable but date_or_birth is in snake_case as attributes.